src/shared/remoteSession.ts

```diff
@@ -18,6 +23,8 @@ export async function openRemoteTerminal(options: vscode.TerminalOptions, onClos
 
         terminal.show()
     })
+
+    timeout.cancel()
```

Comment:
It's good practice to dispose of things in a `finally` block/callback:
```suggestion
    }).finally(() => timeout.cancel())
```

Reply:
Is the advantage of placing this into a finally block that it will run regardless of whether or not the main body throws an exception or not?

Reply:
Yes. Anything that you want to happen regardless of the outcome should go in `finally`.